dask/dataframe/rolling.py

```diff
@@ -57,3 +57,78 @@ def rolling(arg, window, *args, **kwargs):
 rolling_quantile = wrap_rolling(pd.rolling_quantile)
 rolling_apply = wrap_rolling(pd.rolling_apply)
 rolling_window = wrap_rolling(pd.rolling_window)
+
+def call_pandas_rolling_method_single(this_partition, rolling_kwargs,
+        method_name, method_args, method_kwargs):
+    # used for the start of the df/series
+    if this_partition.shape[0] < rolling_kwargs['window']:
+        raise NotImplementedError("Window larger than partition size")
+
+    method = getattr(this_partition.rolling(**rolling_kwargs), method_name)
+    return method(*method_args, **method_kwargs)
+
+def call_pandas_rolling_method_with_neighbor(prev_partition, this_partition,
+        rolling_kwargs, method_name, method_args, method_kwargs):
+    # used for everything except for the start
+
+    window = rolling_kwargs['window']
+    if prev_partition.shape[0] < window:
+        raise NotImplementedError("Window larger than partition size")
+
+    if window > 1:
+        extra = window - 1
+        combined = pd.concat([prev_partition.iloc[-extra:], this_partition])
+
+        method = getattr(combined.rolling(window), method_name)
+        applied = method(*method_args, **method_kwargs)
+        return applied.iloc[extra:]
+    else:
+        method = getattr(this_partition.rolling(window), method_name)
+        return method(*method_args, **method_kwargs)
+
+def wrap_rolling_method(name):
+    def inner(self, *args, **kwargs):
+        return self._call_method(name, *args, **kwargs)
+    return inner
```

Comment:
I'm curious to see how well this will work when serialized.  Dynamically generated functions are sometimes odd.

One option would be to use `toolz.curry` to build these from an `inner` function that was in the main namespace.  Also, the inner function should probably have a better name.  It will be used within diagnostics.

```python
def rolling_part(...):  # previously called inner
    ...

sum = curry(rolling_part, 'sum')
```

This is just one option though, and probably not the best one.